src/middleware/Middleware.sol

```diff
@@ -428,20 +458,40 @@ contract Middleware is SimpleKeyRegistry32, Ownable {
                 continue;
             }
 
-            uint256 operatorStake = getOperatorStake(operator, epoch);
-            totalStake += operatorStake;
+            (uint256 vaultIdx, address[] memory _vaults) = _getOperatorVaults(operator, epochStartTs);
+            assembly {
+                mstore(_vaults, vaultIdx)
+            }
+            if (vaultIdx > 0) {
+                operatorVaultPairs[valIdx++] = OperatorVaultPair(operator, _vaults);
+            }
         }
     }
 
-    /**
-     * @dev Checks if an entity was active at a specific timestamp
-     * @param enabledTime Time when entity was enabled
-     * @param disabledTime Time when entity was disabled (0 if never disabled)
-     * @param timestamp Timestamp to check activity for
-     * @return bool True if entity was active at timestamp
-     */
-    function _wasActiveAt(uint48 enabledTime, uint48 disabledTime, uint48 timestamp) private pure returns (bool) {
-        return enabledTime != 0 && enabledTime <= timestamp && (disabledTime == 0 || disabledTime >= timestamp);
+    function _getOperatorVaults(
+        address operator,
+        uint48 epochStartTs
+    ) private view returns (uint256 vaultIdx, address[] memory _vaults) {
+        _vaults = new address[](s_vaults.length());
+        vaultIdx = 0;
+        for (uint256 j; j < s_vaults.length(); ++j) {
+            (address vault, uint48 vaultEnabledTime, uint48 vaultDisabledTime) = s_vaults.atWithTimes(j);
+
+            // just skip the vault if it was enabled after the target epoch or not enabled
+            if (!_wasActiveAt(vaultEnabledTime, vaultDisabledTime, epochStartTs)) {
+                continue;
+            }
+            uint256 operatorStake = 0;
+            for (uint96 k = 0; k < s_subnetworksCount; ++k) {
+                operatorStake += IBaseDelegator(IVault(vault).delegator()).stakeAt(
```

Comment:
overflow?

Reply:
Don't think it's economically/math feasible. But it's how Symbiotic does it too

Reply:
Also we have a network max limit, so there is no possibility for an overflow if the network doesn't allow it